webapp/public/js/domjudge.js

```diff
@@ -297,69 +325,99 @@ function toggle(id, show)
+    return heartCol.innerHTML + "<span class=\"heart " + iconClass + "\" onclick=\"toggle(" + id + "," + (isFav ? "false" : "true") + "," + mobile + ")\"></span>";
 }
 
 function initFavouriteTeams()
 {
-    var scoreboard = getScoreboard();
-    if (scoreboard === null) {
+    const scoreboards = getScoreboards();
+    if (scoreboards === null) {
         return;
     }
 
     var favTeams = getSelectedTeams();
-    var toAdd = new Array();
-    var cntFound = 0;
-    var lastRank = 0;
-    for (var j = 0; j < scoreboard.length; j++) {
-        var found = false;
-        var teamname = getTeamname(scoreboard[j]);
-        if (teamname === null) {
-            continue;
-        }
-        var firstCol = getRank(scoreboard[j]);
-        var heartCol = getHeartCol(scoreboard[j]);
-        var rank = firstCol.innerHTML;
-        for (var i = 0; i < favTeams.length; i++) {
-            if (teamname === favTeams[i]) {
-                found = true;
-                heartCol.innerHTML = addHeart(rank, scoreboard[j], j, found);
-                toAdd[cntFound] = scoreboard[j].cloneNode(true);
-                if (rank.trim().length === 0) {
-                    // make rank explicit in case of tie
-                    getRank(toAdd[cntFound]).innerHTML += lastRank;
+    Object.keys(scoreboards).forEach(function(key) {
+        var toAdd = new Array();
+        var toAddMobile = new Array();
+        var cntFound = 0;
+        var lastRank = 0;
+        const scoreboard = scoreboards[key];
+        const mobile = key === 'mobile';
+        let teamIndex = 1;
+        for (var j = 0; j < scoreboard.length; j++) {
+            var found = false;
+            var teamname = getTeamname(scoreboard[j]);
+            if (teamname === null) {
+                continue;
+            }
+            var firstCol = getRank(scoreboard[j]);
+            var heartCol = getHeartCol(scoreboard[j]);
+            var rank = firstCol.innerHTML;
+            for (var i = 0; i < favTeams.length; i++) {
+                if (teamname === favTeams[i]) {
+                    found = true;
+                    heartCol.innerHTML = addHeart(rank, scoreboard[j], teamIndex, found, mobile);
+                    toAdd[cntFound] = scoreboard[j].cloneNode(true);
+                    if (mobile) {
+                        toAddMobile[cntFound] = scoreboard[j + 1].cloneNode(true);
+                    }
+                    if (rank.trim().length === 0) {
+                        // make rank explicit in case of tie
+                        getRank(toAdd[cntFound]).innerHTML += lastRank;
+                    }
+                    scoreboard[j].style.background = "lightyellow";
+                    const scoretn = scoreboard[j].querySelector('.scoretn');
+                    if (scoretn && scoretn.classList.contains('cl_FFFFFF')) {
+                        scoretn.classList.remove('cl_FFFFFF');
+                        scoretn.classList.add('cl_FFFFE0');
```

Comment:
Well, I guess that's one way to fix the white colour :joy: 

Is the extra `cl_FFFFE0` class needed, when the wrapping `<div>` will receive the background colour `lightyellow` anyway? :slightly_smiling_face:

Reply:
Yes, that cl_ thing is there because if you set a fixed team column width, we do a 'fade out'. For that to work we need an actual background colour always. So I'm just switching it out. Not the most beautiful but it works.